generator/internal/descriptor_utils.cc

```diff
@@ -634,330 +634,337 @@ bool CheckParameterCommentSubstitutions() {
 
 /// If a service name mapping exists, return the new name.
 /// Parses a command line argument in the form:
-/// {"service_name_mappings": "service_a:new_service_a,service:new_service"}.
+/// {"service_name_mappings": "service_a=new_service_a,service=new_service"}.
 std::string GetEffectiveServiceName(VarsDictionary const& vars,
-                                    std::string const& name) {
+                                    absl::string_view name) {
   auto service_name_mappings = vars.find("service_name_mappings");
-  if (service_name_mappings == vars.end()) return name;
-
-  std::vector<std::string> mappings =
-      absl::StrSplit(service_name_mappings->second, ',');
-  for (auto& mapping : mappings) {
-    std::vector<std::string> kv = absl::StrSplit(mapping, ':');
-    if (kv.size() != 2) continue;
-    if (kv[0] == name) return kv[1];
+  if (service_name_mappings != vars.end()) {
+    std::map<absl::string_view, absl::string_view> map;
+    for (absl::string_view arg :
+         absl::StrSplit(service_name_mappings->second, ',')) {
+      map.insert(absl::StrSplit(arg, absl::MaxSplits('=', 1)));
+    }
+    auto result = map.find(name);
+    if (result != map.end()) {
+      return std::string(result->second.data(), result->second.size());
+    }
+    return std::string(name.data(), name.size());
```

Comment:
In the future, when there is explicit `string_view` to `string` conversion, these will be a little nicer.